### PR TITLE
gh/workflows: Add DP CI for encryption

### PR DESCRIPTION
**Pull request by brb**
:pizza: 

Successful CI run - https://github.com/cilium/cilium/actions/runs/3675330230/jobs/6214617006.

### CODE REVIEW

File: .github/workflows/conformance-datapath.yaml
Comment (pchaigno) [inline .github/workflows/conformance-datapath.yaml]: See https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#expanding-or-adding-matrix-configurations to improve this.

File: .github/workflows/conformance-datapath.yaml
Comment (pchaigno) [inline .github/workflows/conformance-datapath.yaml]: This second install should probably be in its own step with a skip on the matrix.encryption.

File: .github/workflows/conformance-datapath.yaml
Comment (brb) [reply .github/workflows/conformance-datapath.yaml]: Unfortunately, I do not see how the GHA matrix config feature can be useful for this case. The main problem is that each permutation is going to be run on a separate runner in parallel which I want to avoid due to cost reasons.

Comment (pchaigno) [reply .github/workflows/conformance-datapath.yaml]: You are already using a matrix. I'm not suggesting to add more permutations. I'm suggesting to extend the matrix, while keeping the same number of permutations (5), using `include`:
```yaml
strategy:
  matrix:
    include:
      - kernel: 4.19-main
        encryption: ipsec
      - kernel: 5.4-main
        encryption: disabled
      ...
```
You can do that with `kube_proxy` and `kpr` as well of course.

Comment (brb) [reply .github/workflows/conformance-datapath.yaml]: TIL, thanks! Let me try this.

File: .github/workflows/conformance-datapath.yaml
Comment (aanm) [inline .github/workflows/conformance-datapath.yaml:225]: Nit that each one of these lines could be a separate GH workflow step.

File: .github/workflows/conformance-datapath.yaml
Comment (aanm) [inline .github/workflows/conformance-datapath.yaml]: Let's remove this before merge. I've added the dont-merge label. Please remove it once this has been removed as well.

**Review by aanm — APPROVED** [review]

**Comment by brb** [comment]
Currently blocked by https://github.com/cilium/cilium-cli/pull/1299.

**Review by pchaigno — COMMENTED** [review]
LGTM but where is the new link to the successful run? :grin:

**Comment by brb** [comment]
@pchaigno https://github.com/cilium/cilium/actions/runs/3693474477

**Review by pchaigno — APPROVED** [review]

**Pull request merged**
